### ISSUE TITLE
Enable using images from a url or local directory

### ISSUE THREAD

**Issue by AntoniosBarotsis** [AntoniosBarotsis/qr-rs#12]
## TLDR

- [X] Create `--logo-source` param in CLI that can be a link to an image or a path to an image
- [x] Create same param in the server than holds a link to an image
- [X] Make sure that different image formats are handled properly (as in, ~~SVGs~~, JPEGs, PNGs etc should work seamlessly)
- [X] Write tests
- [X] Update [docs](https://antoniosbarotsis.github.io/qr-rs/)
- [x] ~~Switch local assets to SVGs instead~~ This will not be implemented as the image processing library that I am using only supports raster graphics but it also happens that these logos are always very low resolution so SVGs would hardly improve anything.

## Things to consider

~Currently `--logo-source` is only for linking to local files and a separate `--logo-web-source` flag is used for using URLs. This was done to avoid complicating the implementation logic with making sure that the input is a valid link/path and picking the correct response. The name of the flags themselves is up for debate, as are aliases (which are currently non-existent!).~

Ended up using `--path` and `-web` as aliases
 
### Discussed in https://github.com/AntoniosBarotsis/qr-rs/discussions/9

<div type='discussions-op-text'>

<sup>Originally posted by **kbaran1998** January 18, 2023</sup>
## Problem
Currently, it is only possible for the google logo to be the logo in the overlay.

## Possible Solution
It would be nice to be able to point to a directory or a web link to an image for option `-l, --logo <LOGO>` in the CLI and sever. The command could look something like this:

### Local dir
```shell
qrg github.com -d tmp.png -l ./github.png
```

### Web link
```shell
qrg github.com -d tmp.png -l https://cdn-icons-png.flaticon.com/512/25/25231.png
```

</div>
